controller/backup_controller.go

```diff
@@ -444,6 +438,106 @@ func (bc *BackupController) reconcile(backupName string) (err error) {
 	return nil
 }
 
+// handleAttachmentTicketDeletion check and delete attachment so that the source volume is detached if needed
+func (bc *BackupController) handleAttachmentTicketDeletion(backup *longhorn.Backup, volumeName string) (err error) {
+	defer func() {
```

Comment:
nit: Probably I or others had reviewed this. I think basically it deserves a common function/method in the base controller to help with attachment ticket create, delete and verify, then shared with the corresponding controllers.

Reply:
Yes, this will be on [part-2 PR ](https://github.com/longhorn/longhorn/issues/3715#issuecomment-1547290685)coming soon. Thank you